src/components/ciConfig/CIDockerFileConfig.tsx

```diff
@@ -107,46 +101,30 @@ export default function CIDockerFileConfig({
     ]
     const isDefaultBuildContext = (): boolean => {
         if (window._env_.ENABLE_BUILD_CONTEXT) {
-            let currentOverriddenGitMaterialId = 0,
-                currentOverriddenBuildContextGitMaterialId = 0
-            let currentOverriddenBuildContext =
-                ciConfig?.ciPipelines?.[0]?.dockerConfigOverride?.ciBuildConfig?.dockerBuildConfig?.buildContext
-            currentOverriddenGitMaterialId =
-                ciConfig?.ciPipelines?.[0]?.dockerConfigOverride?.ciBuildConfig?.gitMaterialId
-            currentOverriddenBuildContextGitMaterialId =
-                ciConfig?.ciPipelines?.[0]?.dockerConfigOverride?.ciBuildConfig?.buildContextGitMaterialId
+            const currentCIBuildConfig = ciConfig?.ciPipelines?.[0]?.dockerConfigOverride?.ciBuildConfig
+            
+            const currentOverriddenBuildContext = currentCIBuildConfig?.dockerBuildConfig?.buildContext
+            const currentOverriddenGitMaterialId = currentCIBuildConfig?.gitMaterialId
+            const currentOverriddenBuildContextGitMaterialId = currentCIBuildConfig?.buildContextGitMaterialId
+
+            // TODO: Ask whether second check is redundant since string empty comes in ! case
+            const isSameCurrentBuildContext =
+                currentOverriddenGitMaterialId === currentOverriddenBuildContextGitMaterialId &&
+                (!currentOverriddenBuildContext || currentOverriddenBuildContext === '')
```

Comment:
Please remove the `currentOverriddenBuildContext === '` check

Reply:
Done